scratch/evaluate_gaming_traffic.cc

```diff
@@ -25,30 +25,85 @@
 #include "ns3/internet-module.h"
 #include "ns3/point-to-point-module.h"
 #include "ns3/applications-module.h"
-#include "ns3/packet-sink.h"
-#include "ns3/random-variable-stream.h"
 
 using namespace ns3;
 using namespace std;
 
 NS_LOG_COMPONENT_DEFINE ("EvaluateGamingTraffic");
 
+Time delayAccumulator;
+Ptr<GamingStreamingServer> gamingServer;  /* Pointer to the gaming server application */
+Ptr<PacketSink> packetSink;    /* Pointer to the packet sink application */
+uint64_t lastTotalRx = 0;      /* The value of the last total received bytes */
+Time lastPacketTime = Seconds (-1.0);
+
+
+void
+GeneratedPacketsStats (Ptr<const Packet> packet)
+{
+  Time currentPacketTime = Simulator::Now ();
+  if (lastPacketTime.IsStrictlyNegative ())
+    {
+      lastPacketTime = currentPacketTime;
+    }
+  Time packetsInterArrivalTime = currentPacketTime - lastPacketTime;
+  lastPacketTime = currentPacketTime;
+  std::cout << "" << packet->GetSize ()
+            << " " << packetsInterArrivalTime.GetSeconds () << std::endl;
+}
+
+void
+AccumulateDelay (Ptr<const Packet> packet, const Address & addr)
+{
+  TimestampTag timestamp;
+  if (packet->FindFirstMatchingByteTag (timestamp))
+    {
+      Time packetTime = timestamp.GetTimestamp ();
+      delayAccumulator += Simulator::Now () - packetTime;
+    }
+}
+
+void
+CalculateThroughput ()
+{
+  Time now = Simulator::Now ();    /* Return the simulator's virtual time. */
+  double cur = (packetSink->GetTotalRx () - lastTotalRx) * 8.0 / 1e3;     /* Convert Application RX Packets to kb. */
+  std::cout << now.GetSeconds () << "s: \t" << cur << " kbps" << std::endl;
+  lastTotalRx = packetSink->GetTotalRx ();
+  Simulator::Schedule (MilliSeconds (100), &CalculateThroughput);
+}
 
 int
 main (int argc, char *argv[])
 {
+  bool verbose = false;  /* print logging information */
+  bool packetStats = false;  /* print generated packets status */
+  bool summary = false;  /* print application layer traffic summary */
```

Comment:
something should be shown in any case, i would remove this variable and the related condition

Reply:
I'm generating different information like the size of the packet,  IAT, throughput, and logged information about the generated packets. 
by removing these variables how can I control what information should I print?

Reply:
as mentioned in a different comment, you should really print your traces to file, not the the terminal. the terminal should only be used to show some basic information

Reply:
okay, so basically we should save different statistics about the traffic in separate files once the script runs?
for example one file for CDF information, one for throughput and ...
and just present some basic information like as summary?

Reply:
exactly, you can take a look at tommy's PR regarding the new scenario if you're not sure about the code